fe/fe-core/src/main/antlr4/org/apache/doris/nereids/DorisParser.g4

```diff
@@ -966,6 +966,10 @@ sampleMethod
     | INTEGER_VALUE ROWS                                            #sampleByRows
     ;
 
+tableSnapshot
+    : FOR tableSnapshotType=(TIME | VERSION) AS OF valueExpression
```

Comment:
valueExpression is a big scope, such as a + b. If only need number literal, please use `number`. if also need string literal, please use `number | STRING_LITERAL`

Reply:
Good suggesstion! Fixed.
Thanks.